File: UsbSerialConsole/app/src/main/java/jp/sugnakys/usbserialconsole/settings/BasePreferenceFragment.java
package jp.sugnakys.usbserialconsole.settings;

import android.content.SharedPreferences;
import android.preference.ListPreference;
import android.preference.Preference;
import android.preference.PreferenceFragment;
import android.support.v7.widget.Toolbar;
import android.view.View;

import jp.sugnakys.usbserialconsole.R;
import jp.sugnakys.usbserialconsole.util.Log;

public class BasePreferenceFragment extends PreferenceFragment implements SharedPreferences.OnSharedPreferenceChangeListener {

    private static final String TAG = "BasePreferenceFragment";

    SharedPreferences sharedPreference;

    String[] listPrefKeys;

    @Override
    public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
        Preference preference = findPreference(key);
        if (preference instanceof ListPreference) {
            ListPreference listPref = (ListPreference) preference;
            preference.setSummary(listPref.getEntry());
        }
    }

    @Override
    public void onResume() {
        super.onResume();

        sharedPreference = getPreferenceManager().getSharedPreferences();
        sharedPreference.registerOnSharedPreferenceChangeListener(this);
        setSummary();

        Toolbar toolbar = (Toolbar) getActivity().findViewById(R.id.toolbar);
        toolbar.setNavigationOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                getFragmentManager().popBackStack();
            }
        });
    }

    private void setSummary() {
        if (listPrefKeys == null) {
            return;
        }

        ListPreference listPref;
        for (String prefKey : listPrefKeys) {
            listPref = (ListPreference) findPreference(prefKey);
            Log.d(TAG, "Preference: " + prefKey + ", value: " + listPref.getEntry());
            listPref.setSummary(listPref.getEntry());
        }
    }

    @Override
    public void onPause() {
        sharedPreference.unregisterOnSharedPreferenceChangeListener(this);
        super.onPause();
    }
}

File: UsbSerialConsole/app/src/main/java/jp/sugnakys/usbserialconsole/settings/BasePreferenceFragment.kt
package jp.sugnakys.usbserialconsole.settings

import android.content.SharedPreferences
import android.content.SharedPreferences.OnSharedPreferenceChangeListener
import android.preference.ListPreference
import android.preference.PreferenceFragment
import android.view.View
import androidx.appcompat.widget.Toolbar
import jp.sugnakys.usbserialconsole.R
import jp.sugnakys.usbserialconsole.util.Log

open class BasePreferenceFragment : PreferenceFragment(), OnSharedPreferenceChangeListener {
    @JvmField
    var sharedPreference: SharedPreferences? = null

    @JvmField
    var listPrefKeys: Array<String>? = null

    override fun onSharedPreferenceChanged(sharedPreferences: SharedPreferences, key: String) {
        val preference = findPreference(key)
        if (preference is ListPreference) {
            preference.setSummary(preference.entry)
        }
    }

    override fun onResume() {
        super.onResume()
        sharedPreference = preferenceManager.sharedPreferences
        sharedPreference!!.registerOnSharedPreferenceChangeListener(this)
        setSummary()
        val toolbar = activity.findViewById<View>(R.id.toolbar) as Toolbar
        toolbar.setNavigationOnClickListener { fragmentManager.popBackStack() }
    }

    private fun setSummary() {
        if (listPrefKeys == null) {
            return
        }
        var listPref: ListPreference
        for (prefKey in listPrefKeys!!) {
            listPref = findPreference(prefKey) as ListPreference
            Log.d(TAG, "Preference: " + prefKey + ", value: " + listPref.entry)
            listPref.summary = listPref.entry
        }
    }

    override fun onPause() {
        sharedPreference!!.unregisterOnSharedPreferenceChangeListener(this)
        super.onPause()
    }

    companion object {
        private const val TAG = "BasePreferenceFragment"
    }
}